### DISCUSSION TITLE
[help] Understanding format = "file_fast" and also tar_cue(file = F)

### DISCUSSION THREAD

**Discussion by nicki-dese** [ropensci/targets#1083] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Do "convenience" functions that create a file target (eg `tar_files_input`, `tar_file_read`) use `format = "file"` or `format = "file_fast"` when they auto-create the file target?

Is there a way  to force targets to default to `format = "file_fast"` or (better) to revert to the old behaviour of looking at timestamp first for an entire pipeline? I work with very large (several gigabyte) datasets that don't change very often, so the old way was much better for me and my team, this change effects our existing scripts (where we have set format = "file"). 

A related question: if we set `cue = tar_cue(file=F)` option on a target, will it honour or ignore timestamp changes if the file name doesn't change?

Thanks for your excellent work. :)

**Comment by wlandau** [comment, marked as answer]
> Do "convenience" functions that create a file target (eg tar_files_input, tar_file_read) use format = "file" or format = "file_fast" when they auto-create the file target?

Looks like "file_fast" did not make it to all `tarchetypes` functions just yet, I will need to add it.

 > Is there a way to force targets to default to format = "file_fast" or (better) to revert to the old behaviour of looking at timestamp first for an entire pipeline? I work with very large (several gigabyte) datasets that don't change very often, so the old way was much better for me and my team, this change effects our existing scripts (where we have set format = "file").

Sorry, this does require manually changing the format to `"file_fast"`. It is inconvenient, but the new behavior is safer.

> A related question: if we set cue = tar_cue(file=F) option on a target, will it honour or ignore timestamp changes if the file name doesn't change?

`tar_cue(file = FALSE)` will ignore both the time stamp and the hash.
